lib/gooddata/models/user_filters/user_filter_builder.rb

```diff
@@ -140,10 +140,12 @@ def self.verify_existing_users(filters, options = {})
 
     def self.create_label_cache(result, options = {})
       project = options[:project]
+      project_labels = project.labels
+
 
       result.reduce({}) do |a, e|
         e[:filters].map do |filter|
-          a[filter[:label]] = project.labels(filter[:label]) unless a.key?(filter[:label])
+          a[filter[:label]] = (project_labels.find { |l| l.uri == filter[:label] } || project.labels(filter[:label])) unless a.key?(filter[:label])
```

Comment:
Would it be better to eliminate the `||` branch by this? `(project_labels.find { |l| l.uri == filter[:label] || l.id == filter[:label] }`